tests/api/crud/test_secrets.py

```diff
@@ -31,6 +35,415 @@ def test_store_secrets_verifications(
         )
 
 
+def test_store_secrets_with_key_map_verifications(
+    db: sqlalchemy.orm.Session,
+    client: fastapi.testclient.TestClient,
+    k8s_secrets_mock: tests.api.conftest.K8sSecretsMock,
+):
+    project = "project-name"
+    provider = mlrun.api.schemas.SecretProviderName.kubernetes
+    key_map_secret_key = mlrun.api.crud.Secrets().generate_schedule_key_map_secret_key()
+    # not allowed to edit key map
+    with pytest.raises(mlrun.errors.MLRunAccessDeniedError):
+        mlrun.api.crud.Secrets().store_secrets(
+            project,
+            mlrun.api.schemas.SecretsData(
+                provider=provider, secrets={key_map_secret_key: "value"}
+            ),
+        )
+
+    # not allowed with provider other than k8s
+    with pytest.raises(mlrun.errors.MLRunInvalidArgumentError):
+        mlrun.api.crud.Secrets().store_secrets(
+            project,
+            mlrun.api.schemas.SecretsData(
+                provider=mlrun.api.schemas.SecretProviderName.vault,
+                secrets={"invalid/key": "value"},
+            ),
+        )
+
+    # invalid key map name
+    with pytest.raises(mlrun.errors.MLRunInvalidArgumentError):
+        mlrun.api.crud.Secrets().store_secrets(
+            project,
+            mlrun.api.schemas.SecretsData(
+                provider=provider, secrets={"invalid/key": "value"}
+            ),
+            key_map_secret_key="invalid-key-map-secret-key",
+        )
+
+    # internal not allowed
+    with pytest.raises(mlrun.errors.MLRunAccessDeniedError):
+        mlrun.api.crud.Secrets().store_secrets(
+            project,
+            mlrun.api.schemas.SecretsData(
+                provider=provider, secrets={"invalid/key": "value"}
```

Comment:
This doesn't check internal secrets (as the comment hints). Either change the test or the comment...